test/standalone/test-duplicate-nested-expressions.js

```diff
@@ -0,0 +1,198 @@
+/*1* KEEP THIS CODE AT THE TOP TO AVOID LINE NUMBER CHANGES */ /* jshint shadow:true */
+/*2*/'use strict';
+/*3*/function foo(a) {
+/*4*/ var a = 10;
+/*5*/ a += 1;
+/*6*/ return (function (b) {
+/*7*/   var a = 10;
```

Comment:
Could we have the two `a`s in these functions have different values just to make sure the tests are grabbing the correct one?